python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -482,23 +496,13 @@ class RandomForestClassifier(Base):
     def _predict_model_on_gpu(self, X, output_class,
                               threshold, algo,
                               num_classes, convert_dtype):
-        cdef ModelHandle cuml_model_ptr = NULL
         X_m, _, n_rows, n_cols, X_type = \
             input_to_dev_array(X, order='C', check_dtype=self.dtype,
                                convert_to_dtype=(self.dtype if convert_dtype
                                                  else None),
                                check_cols=self.n_cols)
 
-        cdef RandomForestMetaData[float, int] *rf_forest = \
-            <RandomForestMetaData[float, int]*><size_t> self.rf_forest
-
-        build_treelite_forest(& cuml_model_ptr,
-                              rf_forest,
-                              <int> n_cols,
-                              <int> num_classes,
-                              <vector[unsigned char] &> self.model_pbuf_bytes)
-        mod_ptr = <size_t> cuml_model_ptr
-        treelite_handle = ctypes.c_void_p(mod_ptr).value
+        treelite_handle = self._convert_to_treelite(num_classes)
```

Comment:
It would also be nice to have a method that converts a cuML RF model into a FIL model. This method can be placed in RandomForestClassifier/Regressor, ForestInference or both.

Reply:
I'd probably put it in ForestInference - just as FIL can take in a treelite model it could also take in a cuML model. I think it's fine to put that in a separate PR though.

Reply:
I'd prefer it in this PR, as it is well aligned with the theme of the PR, and this PR is fairly small.